src/core/formatter.ml

```diff
@@ -0,0 +1,80 @@
+(* 
+ * Module for formatting user variable
+ * according to the template
+ * DURAND - MARAIS © 2019
+ *)
+
+open Ezjsonm
+open Monad
+
+let value_to_t js = 
+  match js with
+  | `A l -> `A l
+  | `O l -> `O l
+  | _ -> (wrap js)
+
+(* Recover string corresponding to the mustache template *)
+let get_template json =
+  Conf.member json ["template"] 
+  >>= (fun a -> try Ok (get_string a) with _ -> Error "[Error] Template Parse Error") 
+  >>= (fun a -> try Ok (Mustache.of_string a) with _ -> Error "[Error] Template Parser Error")
+
+(* Get only variable declared in the template *)
+let get_template_var json =
+  match json with
+  | `Null | `Bool _ | `Float _ | `String _ | `A _ -> json
+  | `O l -> `O (List.filter (fun (str,_) -> str <> "template") l)
+
+(* Apply function corresponding to [List.map f l] where [f] is a [('a -> 'b choice)] function and [l] is a ['a list]. 
+ * Return a ['b list choice].
+*)
+let map_choice_list f l =
```

Comment:
It's about the choice monad : should be declared into Monad

Reply:
What will be the name of the function?

Reply:
The same :smile:
It's just because it should be into Monad.